wyvern/experimentation/providers/base.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Optional
+
+
+class ExperimentationProvider(str, Enum):
+    EPPO = "eppo"
+
+
+class BaseExperimentationProvider(ABC):
+    """
+    A base class for experimentation providers.
+    All providers should inherit from this and implement the necessary methods.
+    """
+
+    @abstractmethod
+    def get_result(self, experiment_id: str, entity_id: str, **kwargs) -> str:
+        """
+        Get the result (variant) for a given experiment and entity.
+
+        Args:
+        - experiment_id (str): The unique ID of the experiment.
+        - entity_id (str): The unique ID of the entity.
+        - kwargs (dict): Any additional arguments to pass to the provider for targeting.
+
+        Returns:
+        - str: The result (variant) assigned to the entity for the specified experiment.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def log_result(
+        self,
+        experiment_id: str,
+        entity_id: str,
+        variant: Optional[str],
```

Comment:
shall we do `variant: Optional[str] = None`?

[this will break](https://github.com/Wyvern-AI/wyvern/pull/9/files#diff-0287e269080bb1d40817e15963d40bb4325b234944e8c31c0ce3c2a14a53fbd9R66) if `kwargs` doesn't have a `variant` key in it

Reply:
I'll update it but we're passing result as a positional argument in the linked line already

Reply:
ohh i see. that `result` is actually variant. sorry too many params.